File: BSharp/Controllers/DTO/ImportArguments.cs
using BSharp.Controllers.Misc;

namespace BSharp.Controllers.DTO
{
    public class ImportArguments : ParseArguments
    {
        [ChoiceList("Insert", "Update", "Merge", "Delete")]
        public string Mode { get; set; } = "Insert"; // Default
    }
}

File: Tellma/ClientApp/src/app/data/dto/import-arguments.ts
// tslint:disable:variable-name
export interface ImportArguments {
  mode?: ImportMode;
  key?: string;
}

export type ImportMode = 'Insert' | 'Update' | 'Merge';

export const ImportArguments_Mode = {
  Insert: 'Mode_Insert',
  Update: 'Mode_Update',
  Merge: 'Mode_Merge'
};
